### PR TITLE
Removes unused OC_API::register

### PR DESCRIPTION
**Pull request by MorrisJobke**

### CODE REVIEW

File: lib/public/API.php [hunk 1]
```diff
@@ -40,16 +40,6 @@
...
 class API {
-
-	/**
-	 * API authentication levels
-	 * @since 8.1.0
-	 */
-	const GUEST_AUTH = 0;
-	const USER_AUTH = 1;
-	const SUBADMIN_AUTH = 2;
-	const ADMIN_AUTH = 3;
```
Comment (nickvergessen) [inline lib/public/API.php]: still used in the `DeprecationCheck` and the tests for it

Comment (MorrisJobke) [reply lib/public/API.php]: Ummm ... I checked for the usage, and couldn't find this. Let me have a look again.

Comment (MorrisJobke) [reply lib/public/API.php]: Bildschirmfoto 2019-07-18 um 11 41 48

Comment (MorrisJobke) [reply lib/public/API.php]: Found it, let me adjust that then.

**Review by rullzer — APPROVED** [review]
:fire: :bomb: :rocket:

File: lib/public/API.php (hunk 1 above)
Comment (MorrisJobke) [reply lib/public/API.php]: It was the already deleted `OC_API::ADMIN_AUTH` 💡

**Comment by nickvergessen** [comment]
There were 2 failures:

1) Test\App\CodeChecker\CodeCheckerTest::testFindInvalidUsage with data set #3 ('OC_API', 1003, 'test-const.php')

2)  Test\App\CodeChecker\DeprecationCheckTest::testFindInvalidUsage with  data set #4 ('OC_API::ADMIN_AUTH', 1003, 'test-const.php')

**Comment by MorrisJobke** [comment]
> There were 2 failures:
> 
> 1. Test\App\CodeChecker\CodeCheckerTest::testFindInvalidUsage with data set #3 ('OC_API', 1003, 'test-const.php')
> 2. Test\App\CodeChecker\DeprecationCheckTest::testFindInvalidUsage with  data set #4 ('OC_API::ADMIN_AUTH', 1003, 'test-const.php')

Fixed both.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**